README-dev.md

```diff
@@ -1,98 +1,136 @@
+
+    ```text
      eval "$(goenv init -)"
      export PATH="/Users/$USER/.goenv/shims:$PATH"
      ```
    - `goenv install 1.18.10`
    - `goenv global 1.18.10`
-   - Check that the `go version` returns the right version or else you will see the mesage `compile: version "go1.18.10" does not match go tool version "go1.20.2"`. If so, run `brew remove go` or get the matching version.
-8. Invoke `make build`
-  - If you get errors about `libp2p` and `capnp` try with `brew install capnp`
-9. Install language server protocol `opam install ocaml-lsp-server` for better IDE support
-10. Set up your IDE (see [customizing your editor for autocomplete](#customizing-your-dev-environment-for-autocompletemerlin))
+   - Check that the `go version` returns the right version, otherwise you see the message `compile: version "go1.18.10" does not match go tool version "go1.20.2"`. If so, run `brew remove go` or get the matching version.
+8. Invoke `make build`.
+
+  If you get errors about `libp2p` and `capnp`, try with `brew install capnp`.
+9.  For better IDE support, install the OCaml-LSP language server for OCaml: 
+
+  ```sh
+  opam install ocaml-lsp-server
+  ```
+
+10. Set up your IDE. See [Customizing your dev environment for autocomplete/merlin](https://github.com/MinaProtocol/mina/blob/develop/README-dev.md#customizing-your-dev-environment-for-autocompletemerlin).
 
 ### Developer Setup (Linux)
 
 #### Building
 
 Mina has a variety of opam and system dependencies.
 
-To get all the opam dependencies you need, you run `opam switch import opam.export`.
-> *_NOTE:_*  The switch provides a `dune_wrapper` binary that you can use instead of dune, and will fail early if your switch becomes out of sync with the `opam.export` file.
+To get all of the required opam dependencies, run:
 
-Some of our dependencies aren't taken from `opam`, and aren't integrated
-with `dune`, so you need to add them manually, by running `scripts/pin-external-packages.sh`.
+```sh
+opam switch import opam.export
+```
 
-There are a variety of C libraries we expect to be available in the system.
-These are also listed in the dockerfiles. Unlike most of the C libraries,
-which are installed using `apt` in the dockerfiles, the libraries for RocksDB are
-automatically installed when building Mina via a `dune` rule in the library
-ocaml-rocksdb.
+*_NOTE:_*  The `switch` command provides a `dune_wrapper` binary that you can use instead of dune and fails early if your switch becomes out of sync with the `opam.export` file.
+
+Some dependencies that are not taken from `opam` or integrated with `dune` must be added manually. Run the `scripts/pin-external-packages.sh` script.
+
+A number of C libraries are expected to be available in the system and are also listed in the Dockerfiles. Unlike most of the C libraries that are installed using `apt` in the Dockerfiles, the libraries for RocksDB are automatically installed when building Mina by using a `dune` rule in the library `ocaml-rocksdb`.
```

Comment:
Not only must these libraries be available in the system, they also should be in appropriate versions, which is a major headache on most Linux systems, because various distibutions have different versions of libraries available in repositories at each time, depending on their update policy. Actually I tried to build Mina this way in the beginning, but after a week or so my setup broke down after a system update and I gave up, switching to Nix.

Although it has its own difficulties, I think Nix should be presented as the default way of building Mina, while this native build setup should be explicitly labeled as unstable (prone to random breakages) and it should only be proposed as an alternative to experienced Linux users.

Reply:
like my earlier suggestion, can you make changes to satisfy your technical change request? I value your feedback and feel your pain!